### PR TITLE
[FIX] AssayGeneratorMetabo Test  (wo SIRIUS)

### PR DESCRIPTION
**Pull request by oliveralka**
# Description

Due to the introduction of setting the default executable on all platforms, the non SIRIUS related tests failed. 

See Windows CI: https://github.com/OpenMS/OpenMS/runs/3678541178

I am not sure why this did not come up earlier. I guess, we have to add that to release 2.7.0 as well.

https://github.com/OpenMS/OpenMS/pull/5511

Part of current issue: #5541

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Related to the fedora issue. Please link.

**Comment by jpfeuffer** [comment]
But this is not a great user experience to be forced to specify `-executable ""` to make the tool work.
Why is that exactly?

**Comment by oliveralka** [comment]
So, the tool has to options - run with and without SIRIUS.

I switched the run methods to use SIRIUS as primary. That the executable can be found in KNIME automaticcaly, I added the executable sirius.bat/sirius as default. 

Now, when you want to run the tool without SIRIUS, but sirius is set as default, you get the following error:
```
2311: Input file 'sirius' could not be found (by searching on PATH). Either provide a full filepath or fix your PATH environment! Since this file is not strictly required, you might also pass the empty string "" as argument to prevent it's usage (this might limit the usability of the tool).
2311: Error: File not found (the file 'sirius' could not be found) 
```
https://github.com/OpenMS/OpenMS/blob/c2555c3278de34a5f261df9a0915f42a8bfada3c/src/openms/source/APPLICATIONS/TOPPBase.cpp#L1348

So basically this is now handeled by TOPPBase instead of the tool internally, which only wants to know which fragment annotation you use and if the executable is available. 

`else if (use_fragment_annotation && !executable.empty())`
https://github.com/OpenMS/OpenMS/blob/develop/src/utils/AssayGeneratorMetabo.cpp#L434

**Comment by jpfeuffer** [comment]
How about leaving the parameter empty and if sirius is activated, fill it inside the code with the defaults (if it is still empty) and do the executable check (that TOPPBase would do)?

**Comment by jpfeuffer** [comment]
And how about calling it sirius_executable?

**Comment by oliveralka** [comment]
Would that work in KNIME then without specifying the executable?

**Comment by jpfeuffer** [comment]
I don't think this is a KNIME specific issue. If no default executable name is given, it won't find Sirius in any case, right?
The problem is that you have to look in the PATH. And TOPPBase does that automatically.
If the executable is optional, you can of course not require it to be found during startup of the tool.
That means you have to defer the lookup until it is clear that the executable will be needed.

**Comment by oliveralka** [comment]
Ok, thanks for the explanation, then I will try it similar to your suggestion above.

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
Failed test unrelated: `projectroot.src.tests.class_tests.openms.FIAMSScheduler_test`

**Comment by oliveralka** [comment]
@jpfeuffer I guess this has to go to release as well - will open another PR after review.

**Comment by jpfeuffer** [comment]
I am honestly considering a 2.7.1 release instead.

**Comment by oliveralka** [comment]
Ok, will add a branch and open an additional PR.

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins
